src/compat/firestore/firestore.ts

```diff
@@ -197,7 +198,7 @@ export class AngularFirestore {
       collectionRef = pathOrRef;
     }
     const { ref, query } = associateQuery<T>(collectionRef, queryFn);
-    const refInZone = inject(NgZone).run(() => ref);
+    const refInZone = runInInjectionContext(this.injector, () => inject(NgZone).run(() => ref));
```

Comment:
We can just inject the `NgZone` service in the class body instead of running `runInInjectionContext`. I was just not sure if that was going to change the current behavior.

Reply:
Pushed again with this change.